pkgs/development/libraries/tepl/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchurl
+, amtk, gnome3, gtksourceview4, libuchardet, libxml2, pkgconfig }:
+let
+  version = "4.1.1";
+  pname = "tepl";
+in stdenv.mkDerivation {
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "mirror://gnome/sources/${pname}/${gnome3.versionBranch version}/${pname}-${version}.tar.xz";
+    sha256 = "13kflywpc6iyfpc9baaa54in5vzn4p7i3dh9pr2ival2nkxfnkp2";
+  };
+
+  passthru.updateScript = gnome3.updateScript { packageName = pname; };
+
+  nativeBuildInputs = [
+    amtk
```

Comment:
Should these be in native inputs?

Reply:
I really don't know the difference between `buildInputs` and `nativeBuildInputs`... i can try shuffling them around but it seems to build either way.

Reply:
The difference occurs during [cross-compilation](https://nixos.org/nixpkgs/manual/#chap-cross). `nativeBuildInputs` use the architecture of the builder (build platform), while `buildInputs` use the architecture of the host platform (the one the resulting program will be run). That means programs run during build (build managers, compilers, asset generators) need to go to `nativeBuildInputs`.